core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -52,125 +65,123 @@ angular.module('oppia').factory('EditableStoryBackendApiService', [
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
+      ).toPromise().then(
+        response => successCallback(response.status),
+        errorResponse => errorCallback(errorResponse.error));
     };
 
-    var _deleteStory = function(
-        storyId, successCallback, errorCallback) {
-      var storyDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_STORY_DATA_URL_TEMPLATE, {
+    private _deleteStory = function(
+        storyId: string,
+        successCallback: (value?: Object | PromiseLike<Object>) =>void,
+        errorCallback: (reason?: Object | PromiseLike<Object>) => void): void {
+      var storyDataUrl = this.urlInterpolation.interpolateUrl(
+        StoryDomainConstants.EDITABLE_STORY_DATA_URL_TEMPLATE, {
           story_id: storyId
         });
-      $http['delete'](storyDataUrl).then(function(response) {
-        if (successCallback) {
-          successCallback(response.status);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
-    };
 
-    var _doesStoryWithUrlFragmentExist = function(
-        storyUrlFragment, successCallback, errorCallback) {
-      var storyUrlFragmentUrl = UrlInterpolationService.interpolateUrl(
-        STORY_URL_FRAGMENT_HANDLER_URL_TEMPLATE, {
-          story_url_fragment: storyUrlFragment
+      this.http.request(
+        'delete',
+        storyDataUrl).toPromise().then(
+        (response) => {
+          if (successCallback) {
+            successCallback(response.status);
+          }
+        }, function(errorResponse) {
+          if (errorCallback) {
+            errorCallback(errorResponse.error);
+          }
         });
-      $http.get(storyUrlFragmentUrl).then(function(response) {
-        if (successCallback) {
-          successCallback(response.data.story_url_fragment_exists);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
     };
-
-    return {
-      fetchStory: function(storyId) {
-        return $q(function(resolve, reject) {
-          _fetchStory(storyId, resolve, reject);
+    private _doesStoryWithUrlFragmentExist = function(
+        storyUrlFragment: string,
+        successCallback: (value?: Object | PromiseLike<Object>) =>void,
+        errorCallback: (reason?: Object | PromiseLike<Object>) => void): void {
+      var storyUrlFragmentUrl = this.urlInterpolation.interpolateUrl(
+        StoryDomainConstants.STORY_URL_FRAGMENT_HANDLER_URL_TEMPLATE, {
+          story_url_fragment: storyUrlFragment
+        });
+      this.http.get(
+        storyUrlFragmentUrl).toPromise().then(
+        (response) => {
+          if (successCallback) {
+            successCallback(response.status);
+          }
+        }, function(errorResponse) {
+          if (errorCallback) {
+            errorCallback(errorResponse.error);
```

Comment:
`errorResponse.error` -> `errorResponse.error.error` here and elsewhere

Reply:
@nishantwrp done